### ISSUE TITLE
potential memory leak

### ISSUE THREAD

**Issue by Gingeropolous** [moneroexamples/onion-monero-blockchain-explorer#270]
xmrchain runs on a box with 64 gb ram. using top, it shows that the mainnet explorer (the one that gets the most traffic) ends up with 74% memory utilization over time. I occasionally restart it because that amount seems absurd with how clean everything looks and feels with the explorer. After restart, its back at < 5% or so. 

Any thoughts on how I could help hunt this down? Perhaps right around a HF is not the time for this, but just wanted to let you know in case I haven't already.

**Comment by moneroexamples** [comment]
Did it start happening with new changes in the explorer introduced for the upcoming HF?

**Comment by Gingeropolous** [comment]
no, its been going on ever since I've been running xmrchain honestly. I just haven't really been concerned because restarting it "fixes" it, and it seems to run fine, so I didn't think it was critical. 

so its been running since i made that first post above. and once again its at 75% memory.

**Comment by moneroexamples** [comment]
Can you try running it bear bones, without any extra options like `--enable-emission-monitor`, `--enable-autorefresh-option`, ` --enable-pusher `, etc?

**Comment by moneroexamples** [comment]
Ok. I will try to run the explorer and simulate the high traffic, and see how it goes.

**Comment by Gingeropolous** [comment]
well i ran it with only the API, and its currently at 57% ram

**Comment by moneroexamples** [comment]
Still lots. Could you share the settings/options that you use for the explorer?

**Comment by Gingeropolous** [comment]
this is what i normally run:

./xmrblocks -p 8888 \
--enable-pusher 1 \
--enable-key-image-checker 1 \
--enable-output-key-checker 1 \
--enable-autorefresh-option 1 \
--enable-json-api 1 \
--deamon-url http:://127.0.0.1:12345 \
--testnet-url https://testnet.xmrchain.net/ \
--stagenet-url https://stagenet.xmrchain.net/ \
--mainnet-url https://xmrchain.net/ \
--enable-mixin-details 1 \
--enable-emission-monitor 1 \
--enable-randomx


Can I revert my settings to the ones above, or do you still need me to run it minimal?

**Comment by moneroexamples** [comment]
You can run it normally. I will try to investigate what's happening with and without those settings.

**Comment by DaWe35** [comment]
Another possible memory-leak related issue: https://github.com/moneroexamples/onion-monero-blockchain-explorer/issues/308
